src/content/reference/react-dom/components/form.md

```diff
@@ -0,0 +1,451 @@
+
+```js
+<form action={search}>
+    <input name="query" />
+    <button type="submit">Search</button>
+</form>
+```
+
+[See more examples below.](#usage)
+
+#### Props {/*props*/}
+
+`<form>` supports all [common element props.](/reference/react-dom/components/common#props)
+
+* [`action`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#action): a URL or function. When a URL is passed to `action` the form will behave like the HTML form component. When a function is passed to `action` the function will handle the form submission. The function passed to `action` may be async and will be called with a single argument containing the [form data](https://developer.mozilla.org/en-US/docs/Web/API/FormData) of the submitted form. The `action` prop can be overridden by a `formAction` attribute on a `<button>`, `<input type="submit">`, or `<input type="image">` component.
+
+#### Caveats {/*caveats*/}
+
+* When a function is passed to `action` or `formAction` the HTTP method will be POST regaudless of value of the `method` prop.
+
+---
+
+## Usage {/*usage*/}
+
+### Handling form submission without a server {/*handle-form-submission-on-the-client*/}
+
+Render a `<form>` with a input and submit button. Pass a function to the `action` prop of form to run the function when the form is submitted.
+
+<Sandpack>
+
+```js App.js
+export default function Search() {
+  function search(formData) {
+    const query = formData.get("query");
+    alert(`You searched for '${query}'`);
+  }
+  return (
+    <form action={search}>
+      <input name="query" />
+      <button type="submit">Search</button>
+    </form>
+  );
+}
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-6db7f4209-20231021",
+    "react-dom": "18.3.0-canary-6db7f4209-20231021",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js",
+  "devDependencies": {}
+}
+```
+
+</Sandpack>
+
+### Submitting forms without JavaScript {/*submitting-forms-without-javascript*/}
+
+Render a `<form>` with a input and submit button. Pass a function marked with `'use server'` to the `action` prop of form to run the function when the form is submitted. When used with [Server Components](/reference/react/use-client), the form is [progressively enhanced](https://developer.mozilla.org/en-US/docs/Glossary/Progressive_Enhancement).
+
+```jsx
+import { updateCart } from './lib.js';
+
+function AddToCart({productId}) {
+  async function addToCart(formData) {
+    'use server'
+    const productId = formData.get('productId')
+    await updateCart(productId)
+  }
+  return (
+    <form action={addToCart}>
+        <input type="hidden" name="productId" value={productId} />
+        <button type="submit">Add to Cart</button>
+    </form>
+
+  );
+}
+```
+
+In leiu, of using hidden form fields to provide data to the function passed to the `action` prop of `<form>` you can call the `bind` method on the function passed to the `action` prop of `<form>`.
```

Comment:
Oh this took a while for me to understand. Maybe we could set the scene a bit more for the first example

Reply:
Added a note on how hidden form fields can be used to encode data for a form action to consume.